playbooks/prepare_host.yaml

```diff
@@ -37,14 +37,19 @@
         - name: Configure Red Hat Subscription Manager
           ansible.builtin.import_role:
             name: redhat-subscription
+        # this needs to be done early
+        - name: set rhsm_release
+          shell: subscription-manager release --set={{rhsm_release}}
         - name: Install container-tools module # noqa no-changed-when
           ansible.builtin.shell: |
             dnf module disable -y container-tools:rhel8
             dnf module enable -y container-tools:"{{ rhsm_container_tools_version }}"
+          when: rhsm_release | float < 9.0
```

Comment:
we should use `ansible_facts.distribution_major_version`.
are you sure that this module won't exist on OSP 17.1?

Reply:
official docs no longer have this step for rhos 17+ on rhel9+